### PR TITLE
ci: adopt uv as installer

### PR DESCRIPTION
**Pull request by anakin87**
### Related Issues

- similar to https://github.com/deepset-ai/haystack/pull/8313

### Proposed Changes:
- adopt uv as installer

### How did you test it?
CI

### Notes for the reviewer
Some failing tests:
- Chroma tests are failing due to https://github.com/deepset-ai/haystack-core-integrations/issues/1133
- Weaviate tests are failing due to a new release - I'm fixing them in https://github.com/deepset-ai/haystack-core-integrations/pull/1143
- Nvidia tests failing due to Cloud credits expired

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: integrations/amazon_bedrock/pyproject.toml [hunk 1]
```diff
@@ -60,8 +62,9 @@ docs = ["pydoc-markdown pydoc/config.yml"]
 python = ["3.8", "3.9", "3.10", "3.11", "3.12"]
 
 [tool.hatch.envs.lint]
+installer = "uv"
 detached = true
-dependencies = ["black>=23.1.0", "mypy>=1.0.0", "ruff>=0.0.243"]
+dependencies = ["pip", "black>=23.1.0", "mypy>=1.0.0", "ruff>=0.0.243"]
```
Comment (anakin87) [inline integrations/amazon_bedrock/pyproject.toml:67]: `pip` is needed by `mypy` to install missing stub packages

File: integrations/amazon_bedrock/pyproject.toml [hunk 2]
```diff
@@ -42,7 +42,9 @@ root = "../.."
 git_describe_command = 'git describe --tags --match="integrations/amazon_bedrock-v[0-9]*"'
 
 [tool.hatch.envs.default]
+installer = "uv"
 dependencies = [
+  "setuptools",
```
Comment (anakin87) [inline integrations/amazon_bedrock/pyproject.toml:47]: This is needed to generate docs.
If you think it's better, we can add this as a dependency of `haystack-pydoc-tools`.

File: integrations/weaviate/tests/test_document_store.py [hunk 3]
```diff
@@ -265,6 +265,7 @@ def test_to_dict(self, _mock_weaviate, monkeypatch):
                         "session_pool_connections": 20,
                         "session_pool_maxsize": 100,
                         "session_pool_max_retries": 3,
+                        "session_pool_timeout": 5,
```
Comment (anakin87) [inline integrations/weaviate/tests/test_document_store.py:268]: unrelated, needed to fix Weaviate tests and due to https://github.com/weaviate/weaviate-python-client/releases/tag/v4.9.0

File: integrations/amazon_bedrock/pyproject.toml (hunk 2 above)
Comment (anakin87) [reply integrations/amazon_bedrock/pyproject.toml]: done in https://github.com/deepset-ai/haystack-pydoc-tools/pull/5
released new version: https://pypi.org/project/haystack-pydoc-tools/0.3.2/

File: integrations/instructor_embedders/pyproject.toml [hunk 4]
```diff
@@ -30,6 +30,7 @@ dependencies = [
   # Commenting some of them to not interfere with the dependencies of Haystack.
   #"transformers==4.20.0",
   "datasets>=2.2.0",
+  "huggingface_hub<0.26.0",
```
Comment (anakin87) [inline integrations/instructor_embedders/pyproject.toml:33]: Unrelated. https://github.com/xlang-ai/instructor-embedding is practically abandoned and the release of `huggingface_hub` broke it.

File: integrations/chroma/pyproject.toml [hunk 5]
```diff
@@ -61,8 +62,9 @@ docs = ["pydoc-markdown pydoc/config.yml"]
 python = ["3.9", "3.10"]
 
 [tool.hatch.envs.lint]
+installer = "uv"
 detached = true
-dependencies = [
+dependencies = ["pip",
```
Comment (silvanocerza) [inline integrations/chroma/pyproject.toml:67]:
```suggestion
dependencies = [
  "pip", 
```

**Review by silvanocerza — APPROVED** [review]
Tests are failing cause of unrelated changes, will merge in any case.

**Pull request merged**